src/compiler/scala/tools/nsc/typechecker/RefChecks.scala

```diff
@@ -1697,6 +1697,8 @@ abstract class RefChecks extends Transform {
       toConstructor(tree.pos, tree.tpe)
     }
 
+    private lazy val ArrayApplyBoolean = ArrayModule_apply(BooleanTpe)
```

Comment:
Can we use `@deprecatedName` instead?

Reply:
I liked the idea of putting the burden on the lint code, unless the annotation made compelling sense.

Reply:
Array.apply gets the lrytz exclusion. `Boolean*` is not a `Boolean`.